File: 0048-rotate-image/0048-rotate-image.py
class Solution(object):
    def rotate(self, matrix):
        #transpose
        for i in range(len(matrix)):
            for j in range(i):
                matrix[i][j],matrix[j][i]=matrix[j][i],matrix[i][j]
        
        #reverse
        for i in range(len(matrix)):
            matrix[i].reverse()
        """
        :type matrix: List[List[int]]
        :rtype: None Do not return anything, modify matrix in-place instead.
        """
        

File: 0048-rotate-image/0048-rotate-image.java
import java.util.*;
class Solution {
    public void rotate(int[][] matrix) {
        int n = matrix.length; //Row
        int m = matrix[0].length; //Column
        //transpose
        for(int i = 0; i < n; i++){
            for(int j = i ; j< m; j++){
                //swapping matrix[i][j] and matrix[j][i]
                int temp = matrix[i][j];
                matrix[i][j] = matrix[j][i];
                matrix[j][i] = temp;
                
            }             
        }
        
        //reversing matrix
        //traverse row
        for(int i = 0; i < n ; i++){
            //traverse column until middle of row
            for(int j = 0; j < n/2; j++){
                int temp = matrix[i][j];
                matrix[i][j] = matrix[i][n - 1 - j]; //row-same, moving element from opposite end to curr position
                matrix[i][n - 1 - j] = temp;
            }
        }
        
    }
}